lib/types.d.ts

```diff
@@ -0,0 +1,40 @@
+import {IsEqual, ValueOf} from 'type-fest';
+
+type MergeBy<TObject, TOtherObject> = Omit<TObject, keyof TOtherObject> & TOtherObject;
+
+type DeepRecord<TKey extends string | number | symbol, TValue> = {[key: string]: TValue | DeepRecord<TKey, TValue>};
+
+type TypeOptions = MergeBy<
+    {
+        keys: string;
+        collectionKeys: string;
+        values: Record<string, unknown>;
+    },
+    CustomTypeOptions
+>;
+
+interface CustomTypeOptions {}
+
+type Key = TypeOptions['keys'];
+type CollectionKeyBase = TypeOptions['collectionKeys'];
+type CollectionKey = `${CollectionKeyBase}${string}`;
+
+type OnyxKey = Key | CollectionKey;
+type KeyValueMapping = TypeOptions['values'];
+
+type GetOnyxValue<K extends Key | `${CollectionKeyBase}${string}`> = K extends Key
+    ? KeyValueMapping[K]
+    : K extends `${CollectionKeyBase}`
+    ? Record<string, KeyValueMapping[K] | null>
+    : KeyValueMapping[K];
```

Comment:
In this last case, K is not a Key or any kind of collection key, so shouldn't the type here be `undefined` or `null`?

Reply:
@neil-marcellini We are still working on this type, I will add a comment to make it clear.